File: backend/src/main/java/com/collabcloud/model/ActivityLog.java
package com.collabcloud.model;

import javax.persistence.*;
import java.time.LocalDateTime;

@Entity
@Table(name = "activity_logs")
public class ActivityLog {
    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    private Long activityID;

    private String actionType;
    private LocalDateTime timeStamp;

    @ManyToOne(fetch = FetchType.LAZY)
    @JoinColumn(name = "project_id")
    private Project project;

    public ActivityLog() {
    }

    // getters and setters
    public Long getActivityID() {
        return activityID;
    }

    public void setActivityID(Long activityID) {
        this.activityID = activityID;
    }

    public String getActionType() {
        return actionType;
    }

    public void setActionType(String actionType) {
        this.actionType = actionType;
    }

    public LocalDateTime getTimeStamp() {
        return timeStamp;
    }

    public void setTimeStamp(LocalDateTime timeStamp) {
        this.timeStamp = timeStamp;
    }

    public Project getProject() {
        return project;
    }

    public void setProject(Project project) {
        this.project = project;
    }
}

File: frontend/src/models/ActivityLog.ts
// ActivityLog model based on CollabCloud Class Diagram
export interface ActivityLog {
    activityID: string
    actionType: string
    timestamp: Date
}

export class ActivityLogModel implements ActivityLog {
    activityID: string
    actionType: string
    timestamp: Date
    userID: string
    actionDetails?: string

    constructor(activityID: string, userID: string, actionType: string, actionDetails?: string) {
        this.activityID = activityID
        this.userID = userID
        this.actionType = actionType
        this.timestamp = new Date()
        this.actionDetails = actionDetails
    }

    getActivityID(): string {
        return this.activityID
    }

    setActivityID(id: string): void {
        this.activityID = id
    }

    getActionType(): string {
        return this.actionType
    }

    setActionType(type: string): void {
        this.actionType = type
    }

    getActionDetails(): string | undefined {
        return this.actionDetails
    }

    setActionDetails(details: string): void {
        this.actionDetails = details
    }

    getTimestamp(): Date {
        return this.timestamp
    }

    setTimestamp(date: Date): void {
        this.timestamp = date
    }
}
